daemon/containerd/image_pull.go

```diff
@@ -124,6 +125,9 @@ func (i *ImageService) PullImage(ctx context.Context, ref reference.Named, platf
 	img, err := i.client.Pull(ctx, ref.String(), opts...)
 	if err != nil {
 		if errors.Is(err, docker.ErrInvalidAuthorization) {
+			if strings.Contains(err.Error(), "no basic auth credentials") {
```

Comment:
one last nit; can you add a comment with a link to the containerd code, so that we have some reference what we're trying to match (to give it some context)? 

can probably use a permalink to the tag;
https://github.com/containerd/containerd/blob/v1.7.8/remotes/docker/authorizer.go#L189-L191